src/utils/documentUtils.ts

```diff
@@ -40,4 +43,44 @@ const getDocPropsProperties = async (zip: JSZip): Promise<{ doc: Document; prope
     return { doc, properties };
 };
 
-export default { createOrUpdateProperty, getDocPropsProperties };
+const getCellReference = (col: number, row: number): string => {
+    return String.fromCharCode(col + A) + "$" + (row).toString();
+} 
+
+const createCell = (doc: Document, parent: Element, colIndex: number, rowIndex: number, dataType: number, data: string): void => {
+    const cell = doc.createElementNS(doc.documentElement.namespaceURI, "c");
+    cell.setAttribute("r", getCellReference(colIndex, rowIndex + 1).replace("$", ''));
+    const cellData = doc.createElementNS(doc.documentElement.namespaceURI, "v");
+    updateCellData(dataType, data , cell, cellData);
+    cell.appendChild(cellData);
+    parent.appendChild(cell);
+    colIndex++;
+}
+
+const updateCellData = (dataType: number, data: string, newCell: Element, cellData: Element) => {
```

Comment:
As discussed - fact that we are handling overrides in this layer in general means we are carrying user content quite deep, best thing to do is to translate the DTO to a structure you manage internally, typed, and cleaned of malformed data.